getnacelle/nacelle-react
# [examples/nextjs] Update Next.js Preview Mode for Same-Site Preview

### Story

Fixes [DMS-1461](https://nacelle.atlassian.net/browse/DMS-1461)

> Update Next.js example project to support Preview Mode without creating a separate preview site

### Type of Change


### What is being changed and why?

- `examples/nextjs/services/nacelle.js` is updated to:
  - be totally unopinionated about the PIM and CMS that the merchant developer is using
  - allow for preview mode without needing to set up a separate preview site
- `examples/nextjs/pages/api/preview.js` is updated to improve security by requiring a `secret`, and by redirecting to routes in a way that uses Nacelle as a single source of truth, to prevent redirect vulnerabilities



### How to Test



- Check out the `DMS-1461` branch
- `cd examples/nextjs`
- Depending on the CMS source of the Nacelle Space you want to use for testing, follow steps outlined in either the [Shopify + Contentful](https://dms-1461--nacelle-public-docs.netlify.app/next/recipes/preview/preview-shopify-contentful.html) or [Shopify + Sanity](https://dms-1461--nacelle-public-docs.netlify.app/next/recipes/preview/preview-shopify-sanity.html) preview recipe
- Visit some preview routes
  - If using Prairie Wind Apparel, `/products/dania-glasses` is a good example

### Documentation

- https://github.com/getnacelle/nacelle-docs/pull/151 / [dms-1461--nacelle-public-docs.netlify.app](https://dms-1461--nacelle-public-docs.netlify.app/next/intro-next.html)
